### ISSUE TITLE
Results seem to be not least costly; also, feasibility to go from starting point(s) to multiple end points?

### ISSUE THREAD

**Issue by bpfree** [josephlewis/leastcostpath#12]
Trying to create a simple least cost path from 1 starting point to a couple of ending points across a cost raster.

The aim would be to produce results that are similar to what are produced through analyses as from ArcGIS’s [least cost path tool](https://pro.arcgis.com/en/pro-app/latest/tool-reference/intelligence/least-cost-path.htm) or this post from [StackExchange](https://gis.stackexchange.com/questions/28583/gdal-perform-simple-least-cost-path-analysis).

I have all the relevant code and data in this [GitHub](https://github.com/bpfree/gom_cable_analysis). Code for this particular for this analysis is [here](https://github.com/bpfree/gom_cable_analysis/blob/main/code/tentative_analysis/lcp_leastcostpath.R), while data are spread across these three directories:

- [least cost path](https://github.com/bpfree/gom_cable_analysis/tree/main/data/e_least_cost_path)
- [analysis](https://github.com/bpfree/gom_cable_analysis/tree/main/data/c_analysis_data)
- [raster](https://github.com/bpfree/gom_cable_analysis/tree/main/data/d_raster_data)

Some background, there is a single starting location (for simplification) and 4 destination locations (for simplification). Plus there is a cost raster that is the sum of multiple costs. I am working with leastcostpath version 2.0.4:

<img width="792" alt="Screenshot 2022-12-01 at 08 30 13" src="https://user-images.githubusercontent.com/11321624/205065297-0b69b16a-beac-4373-b006-89718599534a.png">

When I plot the generated pathways over the original cost raster, it looks like the lines take a costly path. The darker areas are less costly (lower values) than the lighter ones, yet the path routes over lighter areas.

For the below images pink line are pathways using the create_slope_cs() on my raster, while the cyan lines are using create_cs() on my cost raster.

<img width="687" alt="Screenshot 2022-12-01 at 08 24 08" src="https://user-images.githubusercontent.com/11321624/205064992-e8a27fd3-abc2-454a-86dc-5764d4627d96.png">
<img width="753" alt="Screenshot 2022-12-01 at 08 24 18" src="https://user-images.githubusercontent.com/11321624/205064994-e01ee171-69a2-408e-a7be-bd7ff9654a56.png">
<img width="724" alt="Screenshot 2022-12-01 at 08 24 26" src="https://user-images.githubusercontent.com/11321624/205064998-e19f3494-e7c4-4e67-aa65-adb3feccef67.png">

Two more screenshots of what happens within ArcGIS and how the lines go the expected way.

![image](https://user-images.githubusercontent.com/11321624/205064904-0be31b00-5035-4f53-be0e-0e17fc076629.png)
![image (1)](https://user-images.githubusercontent.com/11321624/205064925-c73ee708-cd45-486c-b575-81fcb842915a.png)

Given this description, is it possible to use this package to get the desired output? If so, any suggestions on where I am going wrong using the package using these data?

**Comment by josephlewis** [comment]
Hi @bpfree,

Thanks for opening the issue.

I will be implementing multiple starting points to multiple end points in the foreseeable future. For now, this can be done by changing the arguments in origin and destination in the create_lcp function

Are you able to create a toy, reproducible example? and/or a single script that I can use to test this. I don't know the structure of the multiple file linked here.

In the meantime, be sure that the surfaces represent the conductivity of movement (leastcostpath uses conductivity rather than cost, i.e. the inverse of cost). This is one of the main differences between ArcGIS and leastcostpath

Hope that helps.

**Comment by bpfree** [comment]
@josephlewis 

Thanks for the prompt reply and thoughtful answers. Good to hear about the functionality for handling multiple starting and/or ending locations. Yes, I have been changing the arguments in the mean time.

As requested, [here](https://github.com/bpfree/gom_cable_analysis/tree/main/leastcostpath_temp/data) is a stripped down version of the analysis (also have attached the directory if that is easier).

Even with conductivity, unless I am understanding it wrong, I would have expected the path to go down to small values (less conductive) and then move along similar values till it reaches the destination. Also when I inspect the path it seems to wriggle more than expected as it moves across the conductanceMatrix.

When I manually inspect the paths against the raster, the paths do not go towards the darker areas -- ones with lower values and where it would be expected to pass through.

Hopefully this has been helpful.

[leastcostpath_temp.zip](https://github.com/josephlewis/leastcostpath/files/10142837/leastcostpath_temp.zip)

**Comment by josephlewis** [comment]
@bpfree Thanks for providing a stripped down version. Very helpful!

If you want to use your already created cost surface (e.g. costs_barriers_extracted) within leastcostpath you'll need to use create_cs. create_slope_cs() is used for digital elevation models with the change in elevation used to calculate a cost (in the case of leastcostpath, time, energy or preference when using wheeled vehicles, or a user-supplied function).

cs <- leastcostpath::create_cs(x = cost_raster, neighbours = 4)

You can view a spatRaster version of the cs using rasterise. This will summarise across all the connections for each cell but it'll give you an idea of what your cs looks like (though in your case it should look exactly the same)

cs_rast <- leastcostpath::rasterise(cs)
plot(cs_rast)

When calculating the LPCs using cs this is what I get. Is this more like what you're after?

![image](https://user-images.githubusercontent.com/22624818/205367444-5b6378bc-dee2-4c5f-9f1c-74744abcdfed.png)

In regards to conductivity, essentially the higher the value the higher the preference for that area (this is in the inverse of cost, where higher cost means lower preference).

**Comment by bpfree** [comment]
Hi @josephlewis apologies for the month delay. Thanks for the reply. Based on the information I realized I just needed to flip my values (low becomes high, and vice versa) and I got the expected results.

![Rplot](https://user-images.githubusercontent.com/11321624/210580617-bdeda1fe-5e55-425e-a8b1-9db1dd81f48e.png)

Also saw that you updated the package to handle multiple starting and ending destinations, which is great! Thanks
